File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        char = ''
        i = 0
        for i in range(0, len(strs[0])):
            for j in range(0, len(strs)): 
                if j == 0: char = strs[j][i]
                else: 
                    if i > (len(strs[j]) -1) or char != strs[j][i]: return strs[0][0:i]
            i+=1
        return strs[0][0:i]

File: 0014-longest-common-prefix/0014-longest-common-prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        if (strs.size == 0) return ""
        val prefixBuilder = StringBuilder()
        var i = 0;
        while (i < strs[0].length) {
            val c = strs[0][i];
            for (s in strs) {
                if (i >= s.length || s[i] != c) {
                    return prefixBuilder.toString()
                }
            }
            prefixBuilder.append(c)
            ++i
        }
        return prefixBuilder.toString()
    }
}